js/modules/k6/html/html.go

```diff
@@ -71,3 +99,438 @@ func (s Selection) Attr(name string, def ...goja.Value) goja.Value {
+	filteredUntil func(string, string) *goquery.Selection,
+	filteredUntilSelection func(string, *goquery.Selection) *goquery.Selection,
+	def ...goja.Value) Selection {
+	// empty selector to nextuntil and prevuntil acts like revAll and nextAll
+	// relies on goquery.compileMatcher returning a matcher which fails all matches when the selector being compiled is invalid
+	if len(def) == 0 {
+		return Selection{s.rt, until("")}
+	}
+
+	selector := def[0].Export()
+	if len(def) == 1 {
+		switch selector.(type) {
+		case string:
+			return Selection{s.rt, until(selector.(string))}
+
+		case Selection:
+			return Selection{s.rt, untilSelection(selector.(Selection).sel)}
+
+		case nil:
+			return Selection{s.rt, until("")}
+
+		default:
+			panic(s.rt.NewGoError(errors.New("Invalid argument. The selector must be a string or query object")))
+			return Selection{}
+		}
+	} else {
+		filter := def[1].String()
+		switch selector.(type) {
+		case string:
+			return Selection{s.rt, filteredUntil(filter, selector.(string))}
+
+		case Selection:
+			return Selection{s.rt, filteredUntilSelection(filter, selector.(Selection).sel)}
+
+		case nil:
+			return Selection{s.rt, filteredUntil(filter, "")}
+
+		default:
+			panic(s.rt.NewGoError(errors.New("Invalid argument. The selector must be a string or query object")))
+			return Selection{}
+		}
+	}
+}
+
+// prevUntil, nextUntil and parentsUntil support two args
+// 1st arg is either a selector string, goquery selection object, or nil
+// 2nd arg is filter selector string or nil/undefined
+func (s Selection) PrevUntil(def ...goja.Value) Selection {
+	return s.adjacentUntil(
+		s.sel.PrevUntil,
+		s.sel.PrevUntilSelection,
+		s.sel.PrevFilteredUntil,
+		s.sel.PrevFilteredUntilSelection,
+		def...,
+	)
+}
+
+func (s Selection) NextUntil(def ...goja.Value) Selection {
+	return s.adjacentUntil(
+		s.sel.NextUntil,
+		s.sel.NextUntilSelection,
+		s.sel.NextFilteredUntil,
+		s.sel.NextFilteredUntilSelection,
+		def...,
+	)
+}
+
+func (s Selection) ParentsUntil(def ...goja.Value) Selection {
+	return s.adjacentUntil(
+		s.sel.ParentsUntil,
+		s.sel.ParentsUntilSelection,
+		s.sel.ParentsFilteredUntil,
+		s.sel.ParentsFilteredUntilSelection,
+		def...,
+	)
+}
+
+func (s Selection) Slice(start int, def ...int) Selection {
+	if len(def) > 0 {
+		return Selection{s.rt, s.sel.Slice(start, def[0])}
+	} else {
+		return Selection{s.rt, s.sel.Slice(start, s.sel.Length())}
+	}
+}
+
+func (s Selection) Get(def ...int) goja.Value {
+	if len(def) == 0 {
+		var items []goja.Value
+
+		for i := 0; i < len(s.sel.Nodes); i++ {
+			items = append(items, selToElement(s.Eq(i)))
+		}
+
+		return s.rt.ToValue(items)
+	} else if def[0] < s.sel.Length() && def[0] > -s.sel.Length() {
+		return selToElement(s.Eq(def[0]))
+	} else {
+		return goja.Undefined()
+	}
+}
+
+func (s Selection) ToArray() (items []Selection) {
+	for i := range s.sel.Nodes {
+		items = append(items, Selection{s.rt, s.sel.Eq(i)})
+	}
+	return
```

Comment:
As cool as it is that you can actually do this, I'd still go with a good ol' `return items` ^^;;